wgpu-core/src/command/bundle.rs

```diff
@@ -440,20 +493,45 @@ impl State {
                 vs.is_dirty = true;
             }
         }
+
+        let push_constants_changed = self
+            .push_constant_ranges
+            .set_push_constants(push_constant_layouts);
+
         self.used_bind_groups = layout_ids.len();
-        let invalid_from = self
-            .bind
-            .iter()
-            .zip(layout_ids)
-            .position(|(bs, layout_id)| match bs.bind_group {
-                Some((_, bgl_id)) => bgl_id != layout_id.value,
-                None => false,
-            });
+        let invalid_from = if push_constants_changed {
+            Some(0)
+        } else {
+            self.bind
+                .iter()
+                .zip(layout_ids)
+                .position(|(bs, layout_id)| match bs.bind_group {
+                    Some((_, bgl_id)) => bgl_id != layout_id.value,
+                    None => false,
+                })
+        };
         if let Some(slot) = invalid_from {
             self.invalidate_group_from(slot);
         }
     }
 
+    fn flush_push_constants(&mut self) -> impl Iterator<Item = RenderCommand> {
+        let is_dirty = self.push_constant_ranges.is_dirty;
+
+        let nonoverlapping_ranges =
+            super::bind::compute_nonoverlapping_ranges(&self.push_constant_ranges.ranges);
+
+        nonoverlapping_ranges
+            .into_iter()
+            .filter(move |_| is_dirty)
```

Comment:
would it be possible to return `Option<impl Iterator<..>>` instead of this hack? :)